src/cf.c

```diff
@@ -84,7 +84,7 @@ const char *CF_GetEncodedChunk(const CuckooFilter *cf, long long *pos, size_t *b
 }
 
 int CF_LoadEncodedChunk(const CuckooFilter *cf, long long pos, const char *data, size_t datalen) {
-    if (datalen == 0) {
+    if (datalen == 0 || pos < datalen + 1) {
```

Comment:
Please add protection for datalen = -1

Reply:
It's type `size_t`, cannot be negative.

Reply:
can be max positive (2^32-1) so (2^32-1+1 is 0)